src/services/fileDestinationUrl.service.ts

```diff
@@ -0,0 +1,49 @@
+import { S3 } from 'aws-sdk';
+import { v4 as uuidv4 } from 'uuid';
+
+const createFileDestinationUrl = async ({
+  bucket,
+  fileType,
+  maxSize = 0,
+  fileName = '',
+  path = '',
+}: {
+  bucket: string;
+  fileType: string;
+  maxSize?: number;
+  fileName?: string;
+  path?: string;
+}): Promise<{
+  destinationUrl: string;
+  presignedPost: S3.PresignedPost;
+}> => {
+  const s3 = new S3();
+  const key = `${path}/${fileName || uuidv4()}`;
+  const s3Params = {
+    Bucket: bucket,
+    Fields: {
+      Key: key,
+    },
+    Expires: 3600,
+    Conditions: [
+      ['eq', '$Content-Type', fileType],
+      ['content-length-range', 0, maxSize],
+    ],
+  };
+
+  const presignedPost: S3.PresignedPost = await new Promise<S3.PresignedPost>((resolve, reject) => {
+    s3.createPresignedPost(s3Params, (err: Error|null, data) => {
+      if (err) reject(err);
+      resolve(data);
+    });
+  });
```

Comment:
Great suggestion!  Moving to a helper method now.

Also just because I did the research originally: AWS SDK actually *does* have a promise based API... but... it is inconsistent.

1. Several AWS sdk methods let you tack on a [`.promise()`](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/using-promises.html) at the end of any method that generates an AWS request, which will convert the call to a promise. 

2. getSignedUrl (which isn't what we are using, since that is for PUT not POST) does not support `.promise` but there is a [`.getSignedUrlPromise()`](https://github.com/aws/aws-sdk-js/pull/2827) which does.

3. `createPresignedPost` does not return an AWS request, so it doesn't have the `.promise()` option, and nobody has added a `getSignedUrlPromise` equivalent (that I could find).

Long story short, ☠️

Reply:
Thanks, and nice research! :skull_and_crossbones: indeed.